### ISSUE TITLE
Cannot mix Accelerated Networking and Non-Accelerated Networking VMs in the same availability set.

### ISSUE THREAD

**Issue by jzampieron** [terraform-providers/terraform-provider-azurerm#782]
This is actually an *undocumented* (at least per the published Microsoft Azure Rest API for VM availability sets) behavior of the Azure REST API. 

If you try to create a VM with Accelerated Networking in an availability set with an existing VM which does not already use AN, then the API throws an error.

```
* azurerm_virtual_machine.machinename: compute.VirtualMachinesClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="ExistingAvailabilitySetWasNotDeployedOnAcceleratedNetworkingEnabledCluster" Message="Cannot add Virtual Machine with accelerated networking-enabled nics (/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/virtualMachines/databaseNodeU0) on an existing non-accelerated networking AvailabilitySet (/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/availabilitySets/databaseVmAvailSet)." Details=[]
```
### Terraform Version
v0.11.1

### Affected Resource(s)
Please list the resources as a list, for example:
- azurerm_virtual_machine
- azurerm_network_interface

### Important Factoids
This ticket is for reference only. It is not likely something that can be fixed in the provider. At least not yet.

**Comment by tombuildsstuff** [comment]
hey @jzampieron 

Thanks for opening this issue :)

Based on the exception message being returned `ExistingAvailabilitySetWasNotDeployedOnAcceleratedNetworkingEnabledCluster` - it appears this should be possible providing the Availability Set is deployed on a cluster which supports Accelerated Networking. It's possible to achieve this either by recreating the Availability Set (as you've seen) or (from memory) by shutting down all machines in the Availability Set and then starting them again. I've opened PR #852 which includes some documentation updates to mention this

Thanks!

**Comment by tombuildsstuff** [comment]
Fixed via #852
